### PR TITLE
nebular-timepicker

### PR DESCRIPTION
**Pull request by violet360**
edit/create event working perfectly (some UI changes needed)
edit and create session form lagging

### CODE REVIEW

**Comment by Ashwin-op** [comment]
@violet360 Just a small suggestion, make sure you either comment or remove the `console.log()` lines

**Comment by violet360** [comment]
> @violet360 Just a small suggestion, make sure you either comment or remove the `console.log()` lines

okk, I will, thank you

File: projects/commudle-admin/src/app/feature-modules/events/components/create-event/create-event.component.ts [hunk 1]
```diff
@@ -45,6 +46,8 @@ export class CreateEventComponent implements OnInit {
       end_date: [''],
       end_hour: [''],
```
Comment (arpansac) [inline projects/commudle-admin/src/app/feature-modules/events/components/create-event/create-event.component.ts]: Is this being used anymore? I think we should remove this?

Comment (arpansac) [reply projects/commudle-admin/src/app/feature-modules/events/components/create-event/create-event.component.ts]: and similarly the start_minute and start_hour

File: projects/commudle-admin/src/app/feature-modules/events/components/create-event/create-event.component.ts [hunk 2]
```diff
@@ -80,15 +85,17 @@ export class CreateEventComponent implements OnInit {
     delete formValue['end_minute'];
     delete formValue['start_date'];
     delete formValue['end_date'];
+    delete formValue['start_time_pick'];
```
Comment (arpansac) [inline projects/commudle-admin/src/app/feature-modules/events/components/create-event/create-event.component.ts:80]: is there a way we can combine all these delete statements??

File: projects/commudle-admin/src/app/feature-modules/events/components/create-event/create-event.component.ts [hunk 3]
```diff
@@ -80,15 +85,17 @@ export class CreateEventComponent implements OnInit {
...
+    delete formValue['end_time_pick'];
 
     if (this.setStartDateTime() && this.setEndDateTime()) {
       if (this.startTime > this.endTime) {
         this.toastLogService.warningDialog('End time has to be greater then start time');
         return
-      }else{
+      }
+      else{
```
Comment (arpansac) [inline projects/commudle-admin/src/app/feature-modules/events/components/create-event/create-event.component.ts]: Why have we divided this into 2 lines?

File: projects/commudle-admin/src/app/feature-modules/events/components/create-event/create-event.component.ts [hunk 4]
```diff
@@ -103,8 +110,9 @@ export class CreateEventComponent implements OnInit {
 
   setStartDateTime() {
     this.startDate = this.eventForm.get('event').get('start_date').value;
-    this.startHour = this.eventForm.get('event').get('start_hour').value;
-    this.startMinute = this.eventForm.get('event').get('start_minute').value;
+    this.startHour = this.eventForm.get('event').get('start_time_pick').value.getHours();
+    this.startMinute = this.eventForm.get('event').get('start_time_pick').value.getMinutes();
+    console.log(this.eventForm.get('event').get('start_time_pick').value.getHours()," ", this.eventForm.get('event').get('start_time_pick').value.getMinutes());
```
Comment (arpansac) [inline projects/commudle-admin/src/app/feature-modules/events/components/create-event/create-event.component.ts]: console.log should not be left in the main code

File: projects/commudle-admin/src/app/feature-modules/events/components/create-event/create-event.component.ts [hunk 5]
```diff
@@ -103,8 +110,9 @@ export class CreateEventComponent implements OnInit {
 
   setStartDateTime() {
     this.startDate = this.eventForm.get('event').get('start_date').value;
-    this.startHour = this.eventForm.get('event').get('start_hour').value;
-    this.startMinute = this.eventForm.get('event').get('start_minute').value;
+    this.startHour = this.eventForm.get('event').get('start_time_pick').value.getHours();
+    this.startMinute = this.eventForm.get('event').get('start_time_pick').value.getMinutes();
```
Comment (arpansac) [inline projects/commudle-admin/src/app/feature-modules/events/components/create-event/create-event.component.ts]: Probably we can store this in a single variable and get hours and minutes from it?

File: projects/commudle-admin/src/app/feature-modules/events/components/create-event/create-event.component.ts [hunk 6]
```diff
@@ -128,8 +136,8 @@ export class CreateEventComponent implements OnInit {
   setEndDateTime() {
 
     this.endDate = this.eventForm.get('event').get('start_date').value;
-    this.endHour = this.eventForm.get('event').get('end_hour').value;
-    this.endMinute = this.eventForm.get('event').get('end_minute').value;
+    this.endHour = this.eventForm.get('event').get('end_time_pick').value.getHours();
+    this.endMinute = this.eventForm.get('event').get('end_time_pick').value.getMinutes();
```
Comment (arpansac) [inline projects/commudle-admin/src/app/feature-modules/events/components/create-event/create-event.component.ts]: same as above

File: projects/commudle-admin/src/app/feature-modules/events/components/edit-event/edit-event.component.html [hunk 7]
```diff
@@ -41,22 +41,11 @@ <h3>Edit Event</h3>
...
-                </div>
-                <div class="minute-select" >
-                  <nb-select fullWidth placeholder="Minute" size="medium" formControlName="start_minute">
-                    <nb-option>NA</nb-option>
-                    <nb-option *ngFor="let i of minutes" [value]="i">
-                      {{ i }}
-                    </nb-option>
-                  </nb-select>
+                <div>
+                  <input nbInput type="text" [formControl]="formControlA" [nbTimepicker]="timepicker" formControlName="start_time_pick">
```
Comment (arpansac) [inline projects/commudle-admin/src/app/feature-modules/events/components/edit-event/edit-event.component.html]: formcontrolA ??
Please pick a proper name

File: projects/commudle-admin/src/app/feature-modules/events/components/edit-event/edit-event.component.html [hunk 8]
```diff
@@ -68,22 +57,13 @@ <h3>Edit Event</h3>
...
-                  <div class="minute-select">
-                    <nb-select fullWidth placeholder="Minute" size="medium" formControlName="end_minute">
-                      <nb-option>NA</nb-option>
-                      <nb-option *ngFor="let i of minutes" [value]="i">
-                        {{ i }}
-                      </nb-option>
-                    </nb-select>
+                <div>
+                  <div>
+                    <input nbInput type="text" [formControl]="formControlB" [nbTimepicker]="timepicker1" formControlName="end_time_pick">
```
Comment (arpansac) [inline projects/commudle-admin/src/app/feature-modules/events/components/edit-event/edit-event.component.html]: formControlB?

File: projects/commudle-admin/src/app/feature-modules/events/components/edit-event/edit-event.component.ts [hunk 9]
```diff
@@ -22,6 +23,9 @@ export class EditEventComponent implements OnInit {
   userTimeZone;
   hours = [...Array(24).keys()];
   minutes = [...Array(60).keys()];
+  // initialDate
+  formControlA;
+  formControlB;
```
Comment (arpansac) [inline projects/commudle-admin/src/app/feature-modules/events/components/edit-event/edit-event.component.ts]: We need better variable names

File: projects/commudle-admin/src/app/feature-modules/events/components/edit-event/edit-event.component.ts [hunk 10]
```diff
@@ -132,8 +143,30 @@ export class EditEventComponent implements OnInit {
...
+    // this.startMinute = this.eventForm.get('event').get('start_minute').value;
+
+    let holdVal = this.eventForm.get('event').get('start_time_pick').value;
+
+    console.log(holdVal);
+
+
+    if(holdVal=="")
+    {
+      console.log("StartingA");
```
Comment (arpansac) [inline projects/commudle-admin/src/app/feature-modules/events/components/edit-event/edit-event.component.ts]: remove

File: projects/commudle-admin/src/app/feature-modules/events/components/edit-event/edit-event.component.ts [hunk 11]
```diff
@@ -157,9 +190,29 @@ export class EditEventComponent implements OnInit {
...
 
-    this.endDate = this.eventForm.get('event').get('start_date').value;
-    this.endHour = this.eventForm.get('event').get('end_hour').value;
-    this.endMinute = this.eventForm.get('event').get('end_minute').value;
+    this.endDate = this.eventForm.get('event').get('end_date').value;
+    // this.endHour = this.eventForm.get('event').get('end_hour').value;
+    // this.endMinute = this.eventForm.get('event').get('end_minute').value;
+
+    let holdVal = this.eventForm.get('event').get('end_time_pick').value;
+    console.log(holdVal);
```
Comment (arpansac) [inline projects/commudle-admin/src/app/feature-modules/events/components/edit-event/edit-event.component.ts]: remove

File: projects/commudle-admin/src/app/feature-modules/events/components/edit-event/edit-event.component.ts [hunk 12]
```diff
@@ -157,9 +190,29 @@ export class EditEventComponent implements OnInit {
...
+    
+    if(holdVal=="")
+    {
+      console.log("endingA");
```
Comment (arpansac) [inline projects/commudle-admin/src/app/feature-modules/events/components/edit-event/edit-event.component.ts]: remove

File: projects/commudle-admin/src/app/feature-modules/events/components/edit-event/edit-event.component.ts [hunk 13]
```diff
@@ -157,9 +190,29 @@ export class EditEventComponent implements OnInit {
...
+      this.endHour = this.formControlB.value.getHours();
+      this.endMinute = this.formControlB.value.getMinutes();
+    }
+
+    if(holdVal!="")
+    {
+      console.log("endingB");
```
Comment (arpansac) [inline projects/commudle-admin/src/app/feature-modules/events/components/edit-event/edit-event.component.ts]: remove

File: projects/commudle-admin/src/app/feature-modules/events/components/edit-event/edit-event.component.ts [hunk 14]
```diff
@@ -157,9 +190,29 @@ export class EditEventComponent implements OnInit {
...
   setEndDateTime() {
 
-    this.endDate = this.eventForm.get('event').get('start_date').value;
-    this.endHour = this.eventForm.get('event').get('end_hour').value;
-    this.endMinute = this.eventForm.get('event').get('end_minute').value;
+    this.endDate = this.eventForm.get('event').get('end_date').value;
+    // this.endHour = this.eventForm.get('event').get('end_hour').value;
+    // this.endMinute = this.eventForm.get('event').get('end_minute').value;
+
+    let holdVal = this.eventForm.get('event').get('end_time_pick').value;
```
Comment (arpansac) [inline projects/commudle-admin/src/app/feature-modules/events/components/edit-event/edit-event.component.ts]: better variable name

File: projects/commudle-admin/src/app/feature-modules/events/components/event-locations/event-location-tracks/event-location-tracks.component.ts [hunk 15]
```diff
@@ -29,6 +31,9 @@ export class EventLocationTracksComponent implements OnInit {
   @ViewChild('tracksContainer') private tracksContainer: ElementRef;
   windowRef;
 
+  formControlA;
+  formControlB;
```
Comment (arpansac) [inline projects/commudle-admin/src/app/feature-modules/events/components/event-locations/event-location-tracks/event-location-tracks.component.ts]: variable names

File: projects/commudle-admin/src/app/feature-modules/events/components/event-locations/event-location-tracks/event-location-tracks.component.ts [hunk 16]
```diff
@@ -122,18 +127,49 @@ export class EventLocationTracksComponent implements OnInit {
 
   showAddSlotForm(eventLocationTrack, hour, minute) {
     this.trackSlotForm.reset();
-    this.trackSlotForm.get('track_slot').patchValue({
+        //     let sTime = moment(this.event.start_time).toDate();
+        // let eTime = moment(this.event.end_time).toDate();
+      console.log("clicked")
```
Comment (arpansac) [inline projects/commudle-admin/src/app/feature-modules/events/components/event-locations/event-location-tracks/event-location-tracks.component.ts]: remove

File: projects/commudle-admin/src/app/feature-modules/events/components/event-locations/event-location-tracks/event-location-tracks.component.ts [hunk 17]
```diff
@@ -159,13 +201,42 @@ export class EventLocationTracksComponent implements OnInit {
...
     this.trackSlotForm.reset();
+    let sTime = trackSlot['start_time'];
+    let eTime = trackSlot['end_time'];
+    let sTimeArr = sTime.split('T')[1].split(':');
+    let eTimeArr = eTime.split('T')[1].split(':');
+    let sTimeHour = parseInt(sTimeArr[0]);
+    let sTimeMinute = parseInt(sTimeArr[1]);
+    let eTimeHour = parseInt(eTimeArr[0]);
+    let eTimeMinute = parseInt(eTimeArr[1]);
+    let o = new Date();
```
Comment (arpansac) [inline projects/commudle-admin/src/app/feature-modules/events/components/event-locations/event-location-tracks/event-location-tracks.component.ts]: o ???

**Review by arpansac — CHANGES_REQUESTED** [review]
Please make all the changes! We really need to pay attention to variable names, indentations

**Pull request merged**